src/main/kotlin/io/github/mojira/arisa/modules/RevokeConfirmationModule.kt

```diff
@@ -0,0 +1,45 @@
+package io.github.mojira.arisa.modules
+
+import arrow.core.Either
+import arrow.core.extensions.fx
+import net.rcarz.jiraclient.ChangeLogEntry
+
+data class RevokeConfirmationModuleRequest(
+    val confirmationStatus: String,
+    val changeLog: List<ChangeLogEntry>
+)
+
+class RevokeConfirmationModule(
+    private val getGroups: (String) -> Either<Throwable, List<String>>,
+    private val setConfirmationStatus: (String) -> Either<Throwable, Unit>,
+    private val confirmationFiledId: String
+) : Module<RevokeConfirmationModuleRequest> {
+    override fun invoke(request: RevokeConfirmationModuleRequest): Either<ModuleError, ModuleResponse> = with(request) {
+        Either.fx {
+            val confirmationChanges = changeLog.filter(::isConfirmationChange)
+            val volunteerChanges = confirmationChanges.filter(::changedByVolunteer)
+            val lastVolunteerChange = volunteerChanges.lastOrNull()
+            val volunteerConfirmation = lastVolunteerChange?.let { getConfirmation(it) } ?: "Unconfirmed"
+
+            assertNotEquals(confirmationStatus, volunteerConfirmation).bind()
+
+            setConfirmationStatus(volunteerConfirmation).toFailedModuleEither().bind()
+        }
+    }
+
+    private fun isConfirmationChange(entry: ChangeLogEntry) =
+        entry.items.any { it.field == confirmationFiledId }
+
+    private fun getConfirmation(entry: ChangeLogEntry) =
+        entry.items.lastOrNull { it.field == confirmationFiledId }?.toString
+
+    private fun changedByVolunteer(entry: ChangeLogEntry): Boolean {
+        val groups = getGroups(entry.author.name)
+
+        return if (groups.isLeft())
+        // when in doubt, assume change was done by a volunteer
+            true
+        else
+            (groups as Either.Right).b.any { it == "helper" || it == "global-moderators" || it == "staff" }
```

Comment:
Not all Mojang staff has the helper, global-moderators or staff role :/

Reply:
Can you give me an example?
To me knowledge, everyone has at least staff.

Reply:
everyone has staff, not all have helper. Perfect